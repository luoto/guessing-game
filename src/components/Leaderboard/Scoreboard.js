import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

class Scoreboard extends Component {
  render() {
    return (
      <div>
        <table>
          <thead>
            <tr>
              <th>Username</th>
              <th>Date</th>
              <th>Difficulty</th>
              <th>Score</th>
            </tr>
          </thead>
          <tbody>
            {this.props.scores.map(score => {
              return (
                <tr key={score.date}>
                  <td>{score.username}</td>
                  <td>{moment(score.date).format('l')}</td>
                  <td>{score.difficulty}</td>
                  <td>{score.score}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    );
  }
}

Scoreboard.propTypes = {
  scores: PropTypes.array.isRequired
};

export default Scoreboard;
