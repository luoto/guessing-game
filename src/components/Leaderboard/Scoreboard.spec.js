import React from 'react';
import { shallow } from 'enzyme';
import Scoreboard from './Scoreboard';

it('renders without crashing', () => {
  shallow(<Scoreboard scores={[]} />);
});

it('renders with 1 row', () => {
  let wrapper = shallow(
    <Scoreboard
      scores={[
        {
          username: 'luoto',
          date: Date.parse('11/25/2018'),
          difficulty: 'medium',
          score: 5
        }
      ]}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
