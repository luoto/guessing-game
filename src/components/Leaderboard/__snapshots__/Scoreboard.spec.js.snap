// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with 1 row 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Username
        </th>
        <th>
          Date
        </th>
        <th>
          Difficulty
        </th>
        <th>
          Score
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="1543132800000"
      >
        <td>
          luoto
        </td>
        <td>
          11/25/2018
        </td>
        <td>
          medium
        </td>
        <td>
          5
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
